File: src/main/java/com/bot/commands/voice/RepeatCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;
import datadog.trace.api.Trace;
import org.springframework.stereotype.Component;

@Component
public class RepeatCommand extends VoiceCommand {

	public RepeatCommand() {
		this.name = "repeat";
		this.arguments = "`all` for repeat all, no input for repeat one";
		this.help = "Toggles repeating the current track or all based on input";
	}

	@Override
	@Trace(operationName = "executeCommand", resourceName = "Repeat")
	protected void executeCommand(CommandEvent commandEvent) {
		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
		if (handler == null) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not currently connected to voice.");
		}
		else {
			if (commandEvent.getArgs().equalsIgnoreCase("all")) {
				handler.setRepeatAll(!handler.isRepeatAll());
				String msg = handler.isRepeatAll() ? "on." : "off.";
				commandEvent.replySuccess("Repeat all is now " + msg);
			} else {
				handler.setRepeatOne(!handler.isRepeatOne());
				String msg = handler.isRepeatOne() ? "on." : "off.";
				commandEvent.replySuccess("Repeat one is now " + msg);
			}
		}
	}
}

File: src/main/java/com/bot/commands/voice/RepeatCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.bot.models.enums.RepeatMode
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class RepeatCommand : VoiceCommand() {
    init {
        name = "repeat"
        arguments = "`all` for repeat all, no input for repeat one"
        help = "Toggles repeating the current track or all based on input"
    }

    @Trace(operationName = "executeCommand", resourceName = "Repeat")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        if (!voiceConnection.isConnected()) {
            commandEvent.reply(commandEvent.client.warning + " I am not currently connected to voice.")
            return
        }
        var mode = RepeatMode.REPEAT_NONE
        if (commandEvent.args.equals("all", ignoreCase = true)) {
           mode = RepeatMode.REPEAT_ALL
        } else if (commandEvent.args.equals("one", ignoreCase = false)) {
            mode = RepeatMode.REPEAT_ONE
        }
        voiceConnection.setRepeatMode(mode)
        commandEvent.reply("${commandEvent.client.success} Set the repeat mode to ${mode.ezName}")

    }
}